nservicebus/upgrades/5to6/distributor.md

```diff
@@ -0,0 +1,87 @@
+---
+title: Deprecated Distributor in Version 6
+tags:
+ - upgrade
+ - migration
+ related:
+ - samples/scaleout/distributor-upgrade
+---
+
+The [distributor](/nservicebus/scalability-and-ha/distributor.md) has been deprecated and replaced by [sender-side distribution]((/nservicebus/msmq/scalability-and-ha/sender-side-distribution.md)) with NServiceBus Version 6.
+
+## Client-side distribution
+
+With sender-side distribution, senders are aware of scaled out receivers, allowing senders to distribute messages directly to the receivers instead of using an intermediary distributor.
+
+## Upgrade scenarios
+
+Based on your availability requirements, there are different approaches available to upgrade endpoints using the distributor to NServiceBus Version 6:
+
+* Upgrade one instance at a time, allowing for high availability scenarios.
+* Upgrade all endpoints communicating with the distributor at the same time.
+
+See the following sections about a detailed description of the described approaches.
+
+
+### One instance at a time
+
+Use this process when shutting down all endpoints at once is no option. This process aims to allow upgrade in production environments at no downtime and without missing events.
+
+* Upgrade all endpoints communicating to the distributor (by commands or event subscription messages) to NServiceBus Version 6 first.
+* Apply the following steps for each scaled out instance enlisted to the Distributor, one instance after another:
+  * Shut down the instance.
+  * [Upgrade the instance to NServiceBus Version 6](#upgrade-endpoint-to-version-6).
+  * [Enlist it with the distributor](#enlist-version-6-endpoints-with-a-distributor).
+  * Start the instance again
+* Once all instances are upgraded to Version 6, configure all endpoints communicating to the distributor (by commands or event subscription messages) to use [sender-side distribution](/nservicebus/msmq/scalability-and-ha/sender-side-distribution.md) to send messages directly to the scaled out instances instead of the distributor.
+* Detach some of the instances (but not all) from the Distributor by removing `endpointConfiguration.EnlistWithLegacyMSMQDistributor` and restarting them.
+* [Unsubscribe the Distributor from all publishers](#remove-subscriptions-for-the-distributor)
```

Comment:
The distributor is usually running on a Windows Failover Cluster because it is a single point of failure. Simply replacing a distributor with a worker would avoid the need to unsubscribe the distributor queue, at first, but it is not really a long term solution. The result would be a worker running on a fail over cluster, which does not make much sense. You'd probably want to remove that from the fail over cluster eventually, forcing you to do the very unsubscribe operation you avoided in the first place anyway.